theano/tensor/nnet/corr_gemm.c

```diff
@@ -172,18 +172,23 @@ PyArrayObject* corrMM(PyArrayObject* bottom,
     }
 
     // Create temporary columns
-    npy_intp col_dim[2];
-    col_dim[0] = (npy_intp)(nChannels * kW * kH);
-    col_dim[1] = (npy_intp)(topHeight * topWidth);
-    PyArrayObject* col = (PyArrayObject*)PyArray_EMPTY(2,
-		                           col_dim,
-                                           PyArray_TYPE(top),
-					   0);
+    int max = 1;
+#if defined(_OPENMP)
+    max = omp_get_max_threads();
+#endif
+    npy_intp col_dim[3];
+    col_dim[0] = (npy_intp)(max);
```

Comment:
@jonhoo, @nouiz, K_ and N_ below still reference col_dim[0] and col_dim[1] below. jonhoo, I made a PR to your branch with fixes.